src/HavenoDaemon.test.ts

```
@@ -465,6 +465,102 @@ test("Can manage Monero daemon connections", async () => {
+    } catch (err) {
+      if (!err.message.startsWith("Failed to start monerod:")) throw new Error("Unexpected error: ");
+    }
+
+    // expect successful start with custom settings
+    let settings: MoneroNodeSettings = new MoneroNodeSettings();
+    let dataDir = rootDir + "/testdata";
+    let logFile = rootDir + "/testdata/test.log";
+    let p2pPort = 38080;
+    let rpcPort = 38081;
+    settings.setBlockchainPath(dataDir);
+    settings.setStartupFlagsList(["--log-file", logFile, "--p2p-bind-port", p2pPort.toString(), "--rpc-bind-port", rpcPort.toString(), "--no-zmq"]);
+    await alice.startMoneroNode(settings);
+    isMoneroNodeRunning = await alice.isMoneroNodeRunning();
+    assert(isMoneroNodeRunning);
+    
+    // expect settings are updated
+    let settingsAfter = await alice.getMoneroNodeSettings();
+    testMoneroNodeSettings(settings, settingsAfter!);
+
+    // expect connection to local monero node to succeed
+    let rpcUrl = "http://localhost:" + rpcPort.toString();
+    let daemon = await monerojs.connectToDaemonRpc(rpcUrl);
+    let height = await daemon.getHeight();
+    assert(height >= 0);
+
+    // expect stopped node
+    await alice.stopMoneroNode();
+    isMoneroNodeRunning = await alice.isMoneroNodeRunning();
+    assert(!isMoneroNodeRunning);
+    try {
+      daemon = await monerojs.connectToDaemonRpc(rpcUrl);
+      height = await daemon.getHeight();
+      throw new Error("should have thrown");
+    } catch (err) {
+      if (err.message !== "RequestError: Error: connect ECONNREFUSED 127.0.0.1:" + rpcPort.toString()) throw new Error("Unexpected error: " + err.message);
+    }
+
+    // start the node back up and continue test
+    await alice.startMoneroNode(settings);
+  }
+  
+  // expect error due to existing running node
+  let beforeSettings = await alice.getMoneroNodeSettings();
+  let newSettings = new MoneroNodeSettings();
+  newSettings.setStartupFlagsList(["--invalid-flag"]); 
```

Surprising the error would indicate the node is already running when an invalid flag is given, but this must be the way monerod is.

---

Right, the flag is unnecessary, I can remove it if it causes confusion since it fails early (node running error before it even tries to start the process with flags)

---

Removed, the flag was an indirect and confusing way to check if the settings were erroneously updated. Testing for error is sufficient.